stable/datadog/templates/daemonset.yaml

```diff
@@ -50,14 +54,22 @@ spec:
         {{- if .Values.datadog.apmEnabled }}
           {{- include "container-trace-agent" . | nindent 8 }}
         {{- end }}
-        {{- if .Values.datadog.processAgentEnabled }}
+        {{- if or (kindIs "invalid" .Values.datadog.processAgentEnabled) .Values.datadog.processAgentEnabled }}
           {{- include "container-process-agent" . | nindent 8 }}
         {{- end }}
-      initContainers:
-        {{ include "containers-init" . | nindent 8 }}
+        {{- if .Values.systemProbe.enabled }}
+          {{- include "system-probe-agent" . | nindent 8 }}
```

Comment:
To be updated to `include "container-system-probe"` after renaming the template.